core/shared/src/main/scala/zio/internal/metrics/ConcurrentMetricRegistry.scala

```diff
@@ -42,6 +44,43 @@ private[zio] class ConcurrentMetricRegistry {
     } else hook0.asInstanceOf[Result]
   }
 
+  def addListener(keyType: MetricKeyType)(listener: MetricListener): Unit =
+    listeners.compute(
+      keyType,
+      { case (_, listeners) =>
+        listeners match {
+          case null      => Set(listener)
+          case listeners => listeners + listener
+        }
+      }
+    )
+
+  def removeListener(listener: MetricListener): Unit =
+    listeners.keySet().forEach { metricKeyType =>
+      listeners.computeIfPresent(metricKeyType, { case (a, b) => b - listener })
+    }
+
+  def update[T](key: MetricKey[MetricKeyType.WithIn[T]], value: T)(implicit trace: Trace): Unit = {
```

Comment:
This is not actually updating, but rather, notifying listeners, so I would rename this to `notifyListenerUpdate` or something.